### PR TITLE
Jarv/update edxapp deploy

### PR DESCRIPTION
**Pull request by jarv**

### CODE REVIEW

File: playbooks/roles/xserver/tasks/deploy.yml [hunk 1]
```diff
@@ -4,32 +4,38 @@
...
   - deploy
 
 - name: create xserver application config
-  template: src=xserver.env.json.j2 dest={{ app_base_dir }}/env.json mode=640 owner=www-data group=adm
+  template: src=xserver.env.json.j2 dest={{app_base_dir}}/env.json mode=640 owner=www-data group=adm
   tags:
   - deploy
 
+- name: install read-only ssh key for the content repo that is required for grading
+  copy: src={{ xserver_git_identity_path }} dest=/etc/git-identity force=yes owner=ubuntu group=adm mode=60
```
Comment (feanil) [inline playbooks/roles/xserver/tasks/deploy.yml:27]: This should get tagged as deploy.

File: playbooks/roles/xserver/tasks/deploy.yml [hunk 2]
```diff
@@ -4,32 +4,38 @@
...
 - name: checkout grader code
-  git: dest={{ xserver_grader_dir }} repo={{ xserver_grader_source }} version={{ xserver_grader_version }}
+  git: dest={{xserver_grader_dir}} repo={{xserver_grader_source}} version={{xserver_grader_version}}
   environment:
     GIT_SSH: /tmp/git_ssh.sh
   tags:
   - deploy
 
+- name: remove read-only ssh key for the content repo
+  file: path=/etc/git-identity state=absent
```
Comment (feanil) [inline playbooks/roles/xserver/tasks/deploy.yml:39]: This should get tagged as deploy.

File: playbooks/roles/edxapp/handlers/main.yml [hunk 3]
```diff
@@ -29,240 +29,11 @@
...
-  # Need to use shell rather than pip so that we can maintain the context of our current working directory; some
-  # requirements are pathed relative to the edx-platform repo. Using the pip from inside the virtual environment implicitly
-  # installs everything into that virtual environment.
-  shell: cd {{ edx_platform_code_dir }} && {{ venv_dir }}/bin/pip install --use-mirrors -r {{ item }}
-  with_items:
-  - "{{ repo_requirements_file }}"
-  - "{{ github_requirements_file }}"
-  - "{{ local_requirements_file }}"
+- name: restart edxapp
+  service: name=edxapp state=restarted
```
Comment (feanil) [inline playbooks/roles/edxapp/handlers/main.yml:21]: I think @jbau said that restart wasn't working correctly for stanford so they used separate start and stop requests.  @jbau did you figure out what was happening here, I was not able to reproduce that.

File: playbooks/roles/xserver/tasks/deploy.yml (hunk 2 above)
Comment (jarv) [inline playbooks/roles/xserver/tasks/deploy.yml:39]: done

File: playbooks/roles/edxapp/handlers/main.yml (hunk 3 above)
Comment (jbau) [inline playbooks/roles/edxapp/handlers/main.yml:21]: I just tried state=restarted and it did work.  Why the change?  _shrug_  But I don't consider it necessary to do stop and then start now.

**Comment by feanil** [comment]
looks good.

**Comment by jarv** [comment]
I also added this sadness - https://github.com/edx/configuration/commit/a8df1341d348555298339a20340c0a85b91623b5

as soon as http://jenkins.edx.org:8080/job/ansible-auto-build/13/console completes I'll merge unless we want to try something different for mysql5.5.

**Comment by feanil** [comment]
sounds good.
